packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -3035,48 +3042,49 @@ class StackViewport extends Viewport {
     viewRef: ViewReference,
     options: ReferenceCompatibleOptions = {}
   ): boolean {
-    if (!super.isReferenceViewable(viewRef, options)) {
-      return false;
-    }
-
-    const { referencedImageId, sliceIndex } = viewRef;
-
-    if (viewRef.volumeId && !referencedImageId) {
-      return options.asVolume;
-    }
-
-    let testIndex = this.getCurrentImageIdIndex();
-    let currentImageId = this.imageIds[testIndex];
-
-    if (options.withNavigation && typeof sliceIndex === 'number') {
-      testIndex = sliceIndex;
-      currentImageId = this.imageIds[testIndex];
-    }
-
-    if (!currentImageId) {
+    const testIndex = this.getCurrentImageIdIndex();
+    const currentImageId = this.imageIds[testIndex];
+    if (!currentImageId || !viewRef) {
       return false;
     }
+    const {
```

Comment:
We should start writing tests for the `isReferenceViewable` feature. Consider adding some for this PR; if not, we should do it in the future.